File: other/Solution80.java
class Solution80 {
    public int removeDuplicates(int[] nums) {
    	if(nums.length <= 2){
    		return nums.length;
    	}
    	int left = 1;
    	int right = 2;
    	for(; right < nums.length; right++){
    		if(nums[left - 1] != nums[right]){
    			left += 1;
    			nums[left] = nums[right];
    		}
    	}
    	return left + 1;
    }

    public static void main(String[] args){
    	Solution80 s = new Solution80();
    	int[] nums = {0,0,1,1,1,1,2,3,3};
    	s.removeDuplicates(nums);
    }
}

File: C++/Solution80.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<stack>
using namespace std;


class Solution80 {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() < 2){
            return nums.size();
        }
        int slow = 1;
        int fast = 2;
        while(fast < nums.size()){
            if(nums[slow - 1] != nums[fast]){
                slow++;
                nums[slow] = nums[fast];
            }
            fast++;
        }
        return slow + 1;
    }
};